### PR TITLE
Implement support for Stacks 2.1 `versioned-smart-contract` transactions

### PR DESCRIPTION
**Pull request by zone117x**
Closes https://github.com/hirosystems/stacks-blockchain-api/issues/1274

The new `versioned-smart-contract` transactions are now being parsed, stored in pg, and the new `clarity_version` property is returned on applicable requests. 

For regular `smart-contract` transactions, the `clarity_version` property is null. 

Note: internally, we are storing this in pg as a new tx type (with the new type ID), but when generating http responses, we normalize the type as `"tx_type": "smart_contract"` so as not to cause unnecessary breaking changes for clients.

####  `GET /extended/v1/tx/<txid>`:
For a `versioned-smart-contract` tx, with a `clarity_version` of `2`:
```json
{
  "tx_id": "0x33c573f5ed06f1feecaa4a9df0225e109416dbba9792abb0cd94869bbad4a88a",
  "nonce": 0,
  "fee_rate": "0",
  "sender_address": "ST000000000000000000002AMW42H",
  "sponsored": false,
  "post_condition_mode": "deny",
  "post_conditions": [],
  "anchor_mode": "any",
  "is_unanchored": false,
  "block_hash": "0xd99feeda9df6cd2019cd48614693501d378d212f57a45649e343539085f15f1b",
  "parent_block_hash": "0x57ee3ea78ff90944bd17696b4037cdf59be69208247bba210b6cc2a61f700727",
  "block_height": 2,
  "burn_block_time": 1660732231,
  "burn_block_time_iso": "2022-08-17T10:30:31.000Z",
  "parent_burn_block_time": 1660732220,
  "parent_burn_block_time_iso": "2022-08-17T10:30:20.000Z",
  "canonical": true,
  "tx_index": 0,
  "tx_status": "success",
  "tx_result": {
    "hex": "0x0703",
    "repr": "(ok true)"
  },
  "microblock_hash": "0x",
  "microblock_sequence": 2147483647,
  "microblock_canonical": true,
  "event_count": 0,
  "events": [],
  "execution_cost_read_count": 0,
  "execution_cost_read_length": 0,
  "execution_cost_runtime": 0,
  "execution_cost_write_count": 0,
  "execution_cost_write_length": 0,
  "tx_type": "smart_contract", <-- Note: this property stays the same to avoid unnecessary breaking changes to clients
  "smart_contract": {
    "clarity_version": 2, <-- NEW
    "contract_id": "ST000000000000000000002AMW42H.pox-2",
    "source_code": ";; PoX testnet constants\n;; Min/max number of reward cycles uSTX can be locked for..."
  }
}
```

For a regular (non-versioned) `smart-contract` tx:

```json
{
  "tx_id": "0x55bb3a37f9b2e8c58905c95099d5fc21aa47d073a918f3b30cc5abe4e3be44c6",
  "nonce": 0,
  "fee_rate": "0",
  "sender_address": "ST000000000000000000002AMW42H",
  "sponsored": false,
  "post_condition_mode": "deny",
  "post_conditions": [],
  "anchor_mode": "any",
  "is_unanchored": false,
  "block_hash": "0xe6afa1531d5e67cd830b7428a31cf9ad5109bc9672ddecdae9f3b32630837bc1",
  "parent_block_hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "block_height": 1,
  "burn_block_time": 1660732814,
  "burn_block_time_iso": "2022-08-17T10:40:14.000Z",
  "parent_burn_block_time": 0,
  "parent_burn_block_time_iso": "",
  "canonical": true,
  "tx_index": 6,
  "tx_status": "success",
  "tx_result": {
    "hex": "0x0703",
    "repr": "(ok true)"
  },
  "microblock_hash": "0x",
  "microblock_sequence": 2147483647,
  "microblock_canonical": true,
  "event_count": 0,
  "events": [],
  "execution_cost_read_count": 0,
  "execution_cost_read_length": 0,
  "execution_cost_runtime": 0,
  "execution_cost_write_count": 0,
  "execution_cost_write_length": 0,
  "tx_type": "smart_contract",
  "smart_contract": {
    "clarity_version": null, <-- NEW
    "contract_id": "ST000000000000000000002AMW42H.bns",
    "source_code": ";;;; Errors\n(define-constant ERR_PANIC 0)..."
  }
}
```

####  `GET /extended/v1/contract/<contract-principal>`:
For a `versioned-smart-contract` tx, with a `clarity_version` of `2`:

```json
{
  "tx_id": "0x33c573f5ed06f1feecaa4a9df0225e109416dbba9792abb0cd94869bbad4a88a",
  "canonical": true,
  "contract_id": "ST000000000000000000002AMW42H.pox-2",
  "block_height": 2,
  "clarity_version": 2, <-- NEW
  "source_code": ";; PoX testnet constants\n;; Min/max number of reward cycles uSTX can be locked for..."
}
```

For a regular (non-versioned) `smart-contract` tx:
```json
{
  "tx_id": "0x55bb3a37f9b2e8c58905c95099d5fc21aa47d073a918f3b30cc5abe4e3be44c6",
  "canonical": true,
  "contract_id": "ST000000000000000000002AMW42H.bns",
  "block_height": 1,
  "clarity_version": null, <-- NEW
  "source_code": ";;;; Errors\n(define-constant ERR_PANIC 0)..."
}
```

TODO:
* [x] Repair tests
* [x] Add tests for ingesting and displaying `versioned-smart-contract` txs

### CODE REVIEW

File: stacks-blockchain/run-local-stacks-node.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (zone117x) [inline stacks-blockchain/run-local-stacks-node.sh:1]: Note: this and other dev environment file changes are temporary until we can get a Stacks 2.1 docker image working correctly.

File: src/datastore/pg-write-store.ts [hunk 2]
```diff
@@ -1167,6 +1172,9 @@ export class PgWriteStore extends PgStore {
       execution_cost_write_count: tx.execution_cost_write_count,
       execution_cost_write_length: tx.execution_cost_write_length,
     };
+    if (tx.type_id === DbTxTypeId.VersionedSmartContract) {
+      console.log('here');
+    }
```
Comment (rafaelcr) [inline src/datastore/pg-write-store.ts:1177]: Remove. Must've slipped by

**Review by rafaelcr — COMMENTED** [review]
LGTM overall, left just one quick comment...

Once the 2.1 docker image is ready we could re-review the launch files and merge soon after. LMK if you'd like me to leave it approved since now, though, fine by me.

**Comment by zone117x** [comment]
Note: I forked `develop` into a `stacks-2.1` branch, which we can merge this PR and the follow-up stacks2.1 related PR into for now.

**Review by rafaelcr — APPROVED** [review]
LGTM

**Pull request merged**